### DISCUSSION TITLE
How Dictu become faster than Clox?

### DISCUSSION THREAD

**Discussion by manoharkakumani** [Jason2605/Dictu#414] [Q&A]
What are the enhancements done to Dictu to make it faster compare to clox ? I've compare with the fibonacci example give in crafting interpreter book. It is way faster than clox even I compared with the Python with Fibonacci example, Dictu is faster than Python.

**Comment by Jason2605** [comment]
It would be interesting to see the results of your testing, especially the Dictu vs CLox comparison. There are a few optimisations included within Dictu that may not be included within clox, such as NaN tagging, computed gotos and a few things in the compiler such as constant folding. I would have expected the difference in performance between the two be pretty negligible though (been a while since i've benchmarked myself in all honesty).

As for the difference between Dictu and (C)Python, I would imagine the speed of Dictu stems from the fact that the VM will be doing a lot less than Python rather than anything technical that has been optimised away.
